vaadin-date-picker-flow/src/main/resources/META-INF/resources/frontend/datepickerConnector.js

```diff
@@ -97,19 +97,29 @@
                 * regex will be the date, so that:
                 * - day-part is '(\d{1,2})' (1 or 2 digits),
                 * - month-part is '(\d{1,2})' (1 or 2 digits),
-                * - year-part is '(\d{4})' (4 digits)
+                * - year-part is '(\d{1,4})' (1 to 4 digits)
                 *
                 * and everything else is left as is.
-                * For example, us date "10/20/2010" => "(\d{1,2})/(\d{1,2})/(\d{4})".
+                * For example, us date "10/20/2010" => "(\d{1,2})/(\d{1,2})/(\d{1,4})".
                 *
                 * The sorting part solves that which part is which (for example,
                 * here the first part is month, second day and third year)
                 *  */
-                datepicker.$connector.regex = testString.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&').replace(datepicker.$connector.dayPart.initial, "(\\d{1,2})").replace(datepicker.$connector.monthPart.initial, "(\\d{1,2})").replace(datepicker.$connector.yearPart.initial, "(\\d{4})");
+                datepicker.$connector.regex = testString.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&')
+                    .replace(datepicker.$connector.dayPart.initial, "(\\d{1,2})")
+                    .replace(datepicker.$connector.monthPart.initial, "(\\d{1,2})")
+                    .replace(datepicker.$connector.yearPart.initial, "(\\d{1,4})");
 
                 datepicker.i18n.formatDate = tryCatchWrapper(function (date) {
-                    let rawDate = new Date(Date.UTC(date.year, date.month, date.day));
-                    return cleanString(rawDate.toLocaleDateString(locale, { timeZone: 'UTC' }));
+                    let rawDate = datepicker._parseDate(`${date.year}-${date.month + 1}-${date.day}`);
+
+                    // Workaround for Safari DST offset issue when using Date.toLocaleDateString().
+                    // This is needed to keep the correct date in formatted result even if Safari
+                    // makes an error of an hour or more in the result with some past dates.
+                    // See https://github.com/vaadin/vaadin-date-picker-flow/issues/126#issuecomment-508169514
+                    rawDate.setHours(12)
```

Comment:
This is needed to avoid reintroducing a bug. See https://github.com/vaadin/vaadin-date-picker-flow/issues/126#issuecomment-508169514

I found this `setHours(12)` to be a simpler workaround in this context without needing to use `Date.UTC` and `{ timeZone: 'UTC' }`. I tested this with iOS 13 Safari. Without this workaround it's impossible to select some dates like `1966-06-01` on iPhone (and desktop Safari) if the system region has been set to e.g. Finland. With this workaround it works.

Would be awesome to have an automated test for this issue but afaik Flow component integration tests are not run with Safari so it's not possible to test with the current setup.

Reply:
I wonder if something like this is an option? It replaces the date values in the formatted sample date so there's no Date object in use at all.

```js
datepicker.i18n.formatDate = tryCatchWrapper(function (date) {
    return testString
        .replace(datepicker.$connector.dayPart.initial, date.day)
        .replace(datepicker.$connector.monthPart.initial, date.month + 1)
        .replace(datepicker.$connector.yearPart.initial, date.year);
});
```

Reply:
The above at least wouldn't automatically add leading zeroes to day and month (when less than 10) which is used in some locales.

Reply:
We could change the day/month values used for the sample to something that have zeros in case required. Like 2 and 3.

Reply:
Note that the suggestion above doesn't work as such also in case `date.day` has the same value as `datepicker.$connector.monthPart.initial`. The second replace would affect the day instead of the month in that case. We would need to use the `part.index` instead of just replace. Not sure if all this would add too much additional logic vs not having to deal with timezones.

Reply:
Imo it would be unnecessary complexity to add such logic compared to current status. This is not dealing with timezones now (it's just local time like it is on the web component too). This is now just setting time to `12.00` for formatting the date preventing off by 1 hour errors in Safari (which can cause wrong date if the time is near midnight).